tools/StatsAnalyzer/Nethermind.StatsAnalyzer.Plugin/Analyzer/Pattern/Builder.cs

```diff
@@ -0,0 +1,79 @@
+namespace Nethermind.PatternAnalyzer.Plugin.Analyzer.Pattern;
+
+public abstract class Builder
+{
+    public abstract PatternStatsAnalyzer Build();
+    public abstract Builder SetSketch(CmSketch sketch);
+    public abstract Builder SetTopN(int topN);
+    public abstract Builder SetMinSupport(ulong support);
+    public abstract Builder SetCapacity(int capacity);
+    public abstract Builder SetBufferSizeForSketches(int size);
+    public abstract Builder SetSketchResetOrReuseThreshold(double error);
+}
+
+public class StatsAnalyzerBuilder : Builder
+{
+    private int? _bufferSizeForSketches;
+    private int? _capacity;
+    private ulong? _minSupport;
+    private CmSketch? _sketch;
+    private double? _sketchResetOrReuseThreshold;
+    private int? _topN;
+
+
+    public override PatternStatsAnalyzer Build()
+    {
+        if (!_bufferSizeForSketches.HasValue)
+            throw new InvalidOperationException("Buffer size for sketches must be set.");
```

Comment:
Minor: you could create helper method `ThrowIfNull<T>(Nullable<T> value, string error)`